src/acl/agent.ts

```diff
@@ -334,7 +334,9 @@ function isAgentAclRule(aclRule: unstable_AclRule): boolean {
  *
  * @param sourceRule ACL rule to duplicate.
  */
-function duplicateAclRule(sourceRule: unstable_AclRule): unstable_AclRule {
+export function duplicateAclRule(
+  sourceRule: unstable_AclRule
+): unstable_AclRule {
```

Comment:
It we're re-using these for groups and classes, we should probably move these to `./acl.ts`, since they're not agent-specific.

Also, given that exporting them makes them available to client libraries, we should prefix their names with `internal_`.

Reply:
Yep - agree they belong better in ./acl.ts.
I thought only functions exported from `index.ts` would be available to client libraries - but exporting anything from anywhere makes it available for client libraries...?

Reply:
Yep. We can restrict that for users of Node 14 and upcoming versions of Webpack (see the ticket on ES Modules), but for every other downstream, everything we export will be available - which is why we're using the `internal_` prefix so often.

Reply:
Moved in `a3de75d`.